### ISSUE TITLE
UnitRegistry lacks user-defined units when evaluating dask arrray

### ISSUE THREAD

**Issue by cbyrohl** [hgrecco/pint#1765]
Using `pint = "0.20.1"`, the following fails

``` python
def test_dask_pint1():
    ureg = pint.UnitRegistry()
    ureg.default_system = 'cgs'
    ureg.define('halfmeter = 0.5 * meter')
    arr = da.ones(1) * ureg("halfmeter")
    print(arr[0].compute().to_base_units())
```

with

``` python
self = <pint.registry.UnitRegistry object at 0x7fb537635ca0>
name_or_alias = 'halfmeter', case_sensitive = None

    def get_name(
        self, name_or_alias: str, case_sensitive: Optional[bool] = None
    ) -> str:
        """Return the canonical name of a unit."""
    
        if name_or_alias == "dimensionless":
            return ""
    
        try:
            return self._units[name_or_alias].name
        except KeyError:
            pass
    
        candidates = self.parse_unit_name(name_or_alias, case_sensitive)
        if not candidates:
>           raise UndefinedUnitError(name_or_alias)
E           pint.errors.UndefinedUnitError: 'halfmeter' is not defined in the unit registry
```

I am a bit puzzled as to which unit registry is being used here as I never defined another and pint documentation states that no global one is created by design. I know that a temporary solution is given by changing the last line to `print(arr[0].to_base_units().compute())` but would like to understand/fix what is happening here.

**Comment by hgrecco** [comment]
I thinks is related to #1749 We well address it in this version.

**Comment by cbyrohl** [comment]
This bug seems to be solved after version 0.20.1 on the main branch, potentially solved by https://github.com/hgrecco/pint/pull/1722/commits/90008e8402bde93b9c3fbb7d884d39f98ee6415c ?
